### PR TITLE
Add timeout for script_run "test -e $inventory"

### PR DESCRIPTION
**Pull request by badboywj**
Avoid sles4sap_publiccloud_basetest post_fail_hook to fail in case of timeout at script_run("test -e $inventory")

- Related ticket: https://jira.suse.com/browse/TEAM-9586
- Needles: N/A
- Verification run: https://openqaworker15.qa.suse.cz/tests/294923#step/Stop_site_a-primary/261

### CODE REVIEW

**Review by lpalovsky — APPROVED** [review]
LGTM

File: lib/qesapdeployment.pm [hunk 1]
```diff
@@ -1273,7 +1273,8 @@ sub qesap_cluster_log_cmds {
 sub qesap_cluster_logs {
     my $provider = get_required_var('PUBLIC_CLOUD_PROVIDER');
     my $inventory = qesap_get_inventory(provider => $provider);
-    if (script_run("test -e $inventory") == 0)
+    my $ret = script_run("test -e $inventory", 90);
+    if ($ret == 0)
```
Comment (alvarocarvajald) [inline lib/qesapdeployment.pm:1277]: No need for the extra `$ret` variable nor to split it in 2 lines if only increasing the timeout.

**Review by alvarocarvajald — APPROVED** [review]
In general LGTM, but I do have a suggestion regarding the changes below, and also, while checking the failure reported in the ticket, I believe a 90s timeout could be enough in some times and not enough in another, as root cause seems to be that the shell session is still busy with the previous command.

File: lib/qesapdeployment.pm (hunk 1 above)
Comment (badboywj) [reply lib/qesapdeployment.pm]: fixed it

**Review by alvarocarvajald — APPROVED** [review]
LGTM

File: lib/qesapdeployment.pm [hunk 2]
```diff
@@ -1273,7 +1273,10 @@ sub qesap_cluster_log_cmds {
 sub qesap_cluster_logs {
     my $provider = get_required_var('PUBLIC_CLOUD_PROVIDER');
     my $inventory = qesap_get_inventory(provider => $provider);
-    if (script_run("test -e $inventory") == 0)
+
+    type_string '', terminate_with => 'ETX';
+    type_string '', terminate_with => 'EOT';
```
Comment (alvarocarvajald) [inline lib/qesapdeployment.pm:1278]: EOT will send a `ctrl-d` ... it will take you out of the shell session.

Comment (badboywj) [reply lib/qesapdeployment.pm]: yes, you are right, I thought to verify the two scenarios with one time run, but it failed, now commented "EOT" line.

**Comment by mpagot** [comment]
What is previous command running that needs CTRL+C? Is it safe to terminate it like this? Generic approach of this particular qesap method is to be as less impacting as possible: get and upload as many logs as possible without perturbing test execution.

**Review by mpagot — APPROVED** [review]
LGTM

**Comment by alvarocarvajald** [comment]
> What is previous command running that needs CTRL+C? Is it safe to terminate it like this? Generic approach of this particular qesap method is to be as less impacting as possible: get and upload as many logs as possible without perturbing test execution.

An ssh command to connect to a VM in the cloud that has not timed out in the shell session, but that has timed out in `script_output` killing the test.

It would be prudent to search every possible place this function is called on and what are the possible previous commands that are run in the shell session. In general a CTRL-C would not do anything if the shell is running and no command is blocking it, but if we want this `test -e inventory.yaml` to work, we need a shell session that is not blocked by anything else.

**Review by mpagot — APPROVED** [review]
LGTM

**Pull request merged**